### PR TITLE
CVE-2010-3256 and CVE-2015-1251

### PR DESCRIPTION
**Pull request by Daniel-J-Fisher**
Starting a pull request for CVE-2010-3256 and CVE-2015-1251 for Dan Fisher

### CODE REVIEW

File: cves/CVE-2010-3256.yml [hunk 1]
```diff
@@ -15,11 +15,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
```
Comment (chtinahow) [inline cves/CVE-2010-3256.yml]: Flesh this description out a little bit more. What are the unspecified impact and attack vectors affected? Also watch for spelling errors.

File: cves/CVE-2010-3256.yml [hunk 2]
```diff
@@ -38,14 +41,16 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: faa43b74c1f27b8f81c0e5d672ae9b62088ce29a
-  :note: ''
+  :note: 'Limit unique entries to 256. Wrtie test.'
```
Comment (chtinahow) [inline cves/CVE-2010-3256.yml]: 256 what? bytes? lines of code? Specify what this means a little bit more

File: cves/CVE-2010-3256.yml [hunk 3]
```diff
@@ -38,14 +41,16 @@ fixes_vcc_instructions: |
...
 - :commit: cf2016d357ca3bc39895c05c691b4def860699e0
-  :note: ''
+  :note: 'Setting a boolean to true when form is submitted.'
 - :commit: 4a103af9e1cced626a6b2b2e77c2e49da29ef619
-  :note: ''
+  :note: 'Wrote some tests.'
```
Comment (chtinahow) [inline cves/CVE-2010-3256.yml]: Tests for the fixed issue or something more unrelated?

File: cves/CVE-2015-1251.yml [hunk 4]
```diff
@@ -15,14 +15,19 @@ description_instructions: |
...
+  Use-after-free vulnerability in the Speech subsystem allows remote attackers 
+  to execute arbitrary code via a crafted document.
+  note
+  git blame 7c730fc5e393d30df80f231571da66d99ccc9e50^ -- third_party/WebKit/Source/modules/speech/SpeechRecognition.cpp
 bounty:
-  date: 2015-05-19 11:49:00.000000000 -04:00
-  amount: 500.0
-  claimed_by: Alexander Bergmann
+  date: 2015-05-20 06:59:03.260000000 -04:00
+  amount:
```
Comment (wadeanthony0100) [inline cves/CVE-2015-1251.yml]: Was there an amount claimed? If not, should there be someone who claimed it?

File: cves/CVE-2015-1251.yml [hunk 5]
```diff
@@ -15,14 +15,19 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: A lot of general bookmark issues seen during that time. Mainly with some
-  issues with redirects as well as ui fixes being made.
+
+description: |
+  Use-after-free vulnerability in the Speech subsystem allows remote attackers 
+  to execute arbitrary code via a crafted document.
```
Comment (wadeanthony0100) [inline cves/CVE-2015-1251.yml]: Minor annoyance-- complete sentences.

File: cves/CVE-2015-1251.yml (hunk 4 above)
Comment (Daniel-J-Fisher) [inline cves/CVE-2015-1251.yml]: So, an amount was offered but the individual declined to accept the payment. Not sure if I should include it in that case.

File: cves/CVE-2010-3256.yml [hunk 6]
```diff
@@ -66,9 +73,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    The code was not tested for this case initially, but the fix commit added
+    a test case to ensure the size would be limited.
+  code: false
```
Comment (andymeneely) [inline cves/CVE-2010-3256.yml]: So, just to clarify - this vulnerability fix inspired them to write unit tests?

File: cves/CVE-2015-1251.yml [hunk 7]
```diff
@@ -15,14 +15,19 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: A lot of general bookmark issues seen during that time. Mainly with some
-  issues with redirects as well as ui fixes being made.
+
+description: |
+  A Use-after-free vulnerability in the Speech subsystem allows remote attackers 
+  to execute arbitrary code via a crafted document.
```
Comment (andymeneely) [inline cves/CVE-2015-1251.yml]: Document? What kinds of documents go into the speech system? File maybe? Audio file? I'm curious what the input was, specifically.

File: cves/CVE-2015-1251.yml [hunk 8]
```diff
@@ -58,9 +65,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    Is it clear the code was not tested before the fix commit. The fix commit
+    contained several tests in CPP as well as front end JS test cases.
```
Comment (andymeneely) [inline cves/CVE-2015-1251.yml]: Not quite the question I had in mind - it's not usual that code: false and fix: true unless they ADDED all of their unit tests in response to this vulnerability. I meant by this question "did they have unit tests to begin with?"

File: cves/CVE-2015-1251.yml [hunk 9]
```diff
@@ -88,8 +100,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: Based on the description in the CVE and the fic commit.
```
Comment (andymeneely) [inline cves/CVE-2015-1251.yml]: "fix"

File: cves/CVE-2015-1251.yml (hunk 7 above)
Comment (Daniel-J-Fisher) [inline cves/CVE-2015-1251.yml]: No one was actual able to craft an exploit to this vulnerability but the following JS is an example of how you could make the speech recognition objec access freed memory. The object was tied to the window contex so when the window is closed, the object is freed.
``` JavaScript
      oWindow = window.open();
      oWebkitSpeechRecognition = new oWindow.webkitSpeechRecognition();
      oWindow.close();
      setInterval(function(){
        if (oWindow.closed) {
          oWebkitSpeechRecognition.start();
        }
      }, 10);
```

File: cves/CVE-2010-3256.yml [hunk 10]
```diff
@@ -38,22 +43,24 @@ fixes_vcc_instructions: |
...
+  :note: 'Do not save data that was submitted through JavaScript.'
+vccs:
+- :commit: 6f2ff852
+  :note: 'The WebDatabase::AddFormFieldValuesTume method never checked for size'
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
```
Comment (mcd4874) [inline cves/CVE-2010-3256.yml]: I would like to give this CVE a 3 vote up

**Pull request merged**